Src/ILGPU/Frontend/CodeGenerator/CodeGenerator.cs

```diff
@@ -82,17 +83,19 @@ private void SetupVariables()
                 {
                     case ILInstructionType.Ldarga:
                         variables.Add(new VariableRef(
-                            instruction.GetArgumentAs<int>(), VariableRefType.Argument));
+                            instruction.GetArgumentAs<int>() - LambdaArgumentOffset,
+                            VariableRefType.Argument));
                         break;
                     case ILInstructionType.Ldloca:
                         variables.Add(new VariableRef(
-                            instruction.GetArgumentAs<int>(), VariableRefType.Local));
+                            instruction.GetArgumentAs<int>() - LambdaArgumentOffset,
```

Comment:
Are you sure about this piece of code? The 'this' argument shouldn't change all local variable indexes in the function, should it?

Reply:
Ah, my mistake, I misread that opcode thinking it was a variation to `Ldarg`. Will fix.

Reply:
Done.